core/geometry/src/curve/internalContexts/geometry2d/UnboundedCircle2d.ts

```diff
@@ -0,0 +1,185 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module CartesianGeometry
+ */
+
+
+import { Geometry } from "../../../Geometry";
+import { Point2d, Vector2d } from "../../../geometry3d/Point2dVector2d";
+import { XAndY } from "../../../geometry3d/XYZProps";
+import { Degree2PowerPolynomial } from "../../../numerics/Polynomials";
+import { ImplicitCurve2d } from "./implicitCurve2d";
+import { UnboundedLine2dByPointAndNormal } from "./UnboundedLine2d.";
+
+/**
+ * Internal class for a complete circle in the xy plane, with center and radius stored.
+ */
+export class UnboundedCircle2dByCenterAndRadius extends ImplicitCurve2d {
+  /** The Cartesian coordinates of any center on the line. */
+  public center: Point2d;
+  /** The circle radius */
+  public radius: number;
+  /* Constructor - CAPTURE given center and normal */
+  private constructor(center: Point2d, radius: number) {
+    super();
+    this.center = center;
+    this.radius = radius;
+  }
+    /**
+     * Return a clone of this circle.
+     */
+    public clone () : UnboundedCircle2dByCenterAndRadius {
+      // (The create method clones the inputs . . .)
+      return  UnboundedCircle2dByCenterAndRadius.createPointRadius (this.center, this.radius);
+    }
+    /**
+     * Return a clone of this circle, with radius negated
+     */
+    public cloneNegateRadius () : UnboundedCircle2dByCenterAndRadius {
+      // (The create method clones the inputs . . .)
+      return  UnboundedCircle2dByCenterAndRadius.createPointRadius (this.center, -this.radius);
+    }
+  /**
+   * Create an ImplicitCircle2d from XY parts of its center and its radius
+   * @param centerX x coordinate of center
+   * @param centerY y coordinate of center
+   * @param radius circle radius
+   * @returns newly created circle object
+   */
+  public static createXYRadius(centerX: number, centerY: number, radius: number): UnboundedCircle2dByCenterAndRadius {
+    return new UnboundedCircle2dByCenterAndRadius(Point2d.create(centerX, centerY), radius);
+  }
+
+  /**
+   * Create an ImplicitCircle2d from an xy object and a radius.
+   * * Zero radius is valid.
+   * * The input coordinates are copied -- the center is NOT captured.
+   * @param center xy coordinates of center
+   * @param radius circle radius
+   * @returns newly created circle object
+   */
+  public static createPointRadius(center: XAndY, radius: number): UnboundedCircle2dByCenterAndRadius {
+    return new UnboundedCircle2dByCenterAndRadius(Point2d.create(center.x, center.y), radius);
+  }
+  /**
+   *
+   * @param xy space paoint
```

Comment:
Fix here and next lines
```suggestion
   * @param xy space point
```